docs/development/development_environment.rst

```diff
@@ -2,31 +2,48 @@ Development Environment
+To correctly set up your development environment for this repository, clone the repository from github, and install the project dependencies.
 
 .. code-block:: bash
 
     git clone https://github.com/waltsims/k-wave-python
     cd k-wave-python
-    pip install -e .
+    pip install -e '.[dev]'
 
-Next, download all k-Wave binaries from the `k-Wave download page <http://www.k-wave.org/download.php>`_.
+Test References
+=======================
 
-Lastly, place the contents of the linux-binaries, and windows-executables directories in the project directory structure under ``kwave/bin/linux/``, ``kwave/bin/darwin`` and ``kwave/bin/windows`` respectively.
-You will have to create the directory structure yourself.
+Test compare the output of the python interface to the output of the matlab interface.
+These tests are located in the ``tests`` directory. The comparison between ``matlab`` and ``python`` outputs are done in two ways:
 
-With this, you are ready to develop k-Wave-python.
-If you have any issues or questions, please post them on the `k-Wave-python discussions page <https://github.com/waltsims/k-wave-python/discussions>`_ to discuss. We look forward to interacting with you.
+- **Unit testing**: k-Wave-python functions that have a direct counterpart in original k-Wave are tested by comparing the outputs of the two functions.
+  The output of the original k-Wave functions are stored in ``.mat`` files.
+  These files can be generated by running the corresponding MATLAB scripts located in the ``tests/matlab_test_data_collectors/matlab_collectors/`` directory by running ``tests/matlab_test_data_collectors/run_all_collectors.m``.
+  After running the scripts, the reference files can be found in ``tests/matlab_test_data_collectors/python_testes/collectedValues/``.
+ 
+.. note::
+    If you do not have MATLAB installed to generate the reference files, you can download recently generated reference file outputs from the GitHub CI and place them in the ``python_testers/collectedValues/`` directory.
+    The latest reference files can be found in the artifacts of the latest CI run of ``pytest.yml``.
 
+- **Integration testing**: k-Wave-python tests output .h5 files that are passed to the k-Wave binaries and ensures that they match the output of the original k-Wave.
+  This testing compares the output for many of the example scripts from the original k-Wave package.
+  Hash values of the reference output ``.h5`` file from MATLAB examples are generated and stored in ``.json`` files in ``tests/reference_outputs/``.
+  This ``.json`` files are contained in the code repository and do not need to be generated.
+  Since these files are generated from the original k-Wave package, they only need to be updated when a new release of k-Wave is made.
 
-Test References
-=======================
+<TODO: @farid How to regenerate theses outputs?>
```

Comment:
@faridyagubbayli Your general feedback is welcome, but please describe how to generate the .json files here.